persistence/mongo/src/main/scala/za/co/absa/spline/persistence/mongo/dao/LineageDAOv4.scala

```diff
@@ -100,6 +100,15 @@ class LineageDAOv4(override val connection: MongoConnection) extends BaselineLin
             addAll(opTransformations.asJava)
           })
         }
+        else if (isWriteOperation(op)) {
+          if (!op.containsField("writeMetrics")) {
+            op.put("writeMetrics", new BasicDBObject(new ju.HashMap()))
```

Comment:
Do you need to add something like this also into MutableLineageUpgraderV4#upgrader?

Reply:
no, because this needs to be executed on v4 objects. The v4Upgrader is called on v3 objects (to upgrade ones to v4)